--- unison-2.52.1/src/Makefile.OCaml.orig	2022-05-07 18:12:13.000000000 +0200
+++ unison-2.52.1/src/Makefile.OCaml	2022-06-06 00:27:40.210613559 +0200
@@ -469,13 +469,13 @@
 
 $(NAME)$(EXEC_EXT): $(CAMLOBJS) $(COBJS)
 	@echo Linking $@
-	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(CFLAGS) $(CAMLLIBS) $^ $(CLIBS)
+	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(OCAML_LINKING_CFLAGS) $(CAMLLIBS) $^ $(CLIBS)
 
 # Unfortunately -output-obj does not put .o files into the output, only .cmx
 # files, so we have to use $(LD) to take care of COBJS.
 $(NAME)-blob.o: $(CAMLOBJS) $(COBJS)
 	@echo Linking $@
-	$(CAMLC) -dstartup -output-obj -verbose -cclib -keep_private_externs $(CAMLFLAGS) -o u-b.o $(CFLAGS) $(CAMLLIBS) $(CLIBS) $(CAMLOBJS)
+	$(CAMLC) -dstartup -output-obj -verbose -cclib -keep_private_externs $(CAMLFLAGS) -o u-b.o $(OCAML_LINKING_CFLAGS) $(CAMLLIBS) $(CLIBS) $(CAMLOBJS)
 	$(LD) -r -keep_private_externs -o $@ u-b.o $(COBJS)
 	$(RM) u-b.o
 
--- unison-2.52.1/src/fsmonitor/linux/Makefile.orig	2022-05-07 18:12:13.000000000 +0200
+++ unison-2.52.1/src/fsmonitor/linux/Makefile	2022-06-06 01:44:46.813674472 +0200
@@ -22,7 +22,7 @@
 
 $(FSMONITOR)$(EXEC_EXT): $(COMPATOCAMLOBJS) $(FSMCAMLOBJS) $(FSMCOBJS)
 	@echo Linking $@
-	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(CFLAGS) $(FSMCAMLLIBS) $^ $(CLIBS)
+	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(OCAML_LINKING_CFLAGS) $(FSMCAMLLIBS) $^ $(CLIBS)
 
 clean::
 	rm -f $(DIR)/*.cm[iox] $(DIR)/*.o $(DIR)/*~
